test/e2e/util/resource_manager.go

```diff
@@ -0,0 +1,316 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	"github.com/operator-framework/operator-lifecycle-manager/test/e2e/ctx"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	k8scontrollerclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+type K8sResourceManager interface {
```

Comment:
IMO, interfaces should be defined on the consumer side. Defining interfaces on the producer side, like this, leads to large interfaces where consumers are forced to depend on APIs they don't need.